bindings_test.go

```diff
@@ -14,46 +14,35 @@ import (
 
 func TestFanout(t *testing.T) {
 	var (
-		assert      = assert.New(t)
 		timesCalled int64
-		called      = make(chan struct{})
 	)
 
 	fanoutHandler := func(ctx context.Context, rw *ResponseWriter, d amqp.Delivery) {
 		atomic.AddInt64(&timesCalled, 1)
-		called <- struct{}{}
 	}
 
 	for range make([]struct{}, 3) {
-		s := NewServer(serverTestURL)
+		s := NewServer(testURL)
 		s.Bind(FanoutBinding("fanout-exchange", fanoutHandler))
 
 		stop := startAndWait(s)
 		defer stop()
 	}
 
-	c := NewClient(serverTestURL)
+	c := NewClient(testURL)
 	defer c.Stop()
 
 	_, err := c.Send(NewRequest().WithExchange("fanout-exchange").WithResponse(false))
+	assert.NoError(t, err)
 
-	// Ensure all handlers have added to the timesCalled variable.
-	for range make([]int, 3) {
-		select {
-		case <-called:
-			// Great!
-		case <-time.After(time.Second):
-			t.Error("fanoutHandler was not called")
-		}
-	}
-
-	assert.Nil(err, "no errors occurred")
-	assert.Equal(atomic.LoadInt64(&timesCalled), int64(3), "endpoint called 3 times")
+	assert.Eventually(t, func() bool {
```

Comment:
Works for me?

Reply:
You mean the test fails after 3 seconds? Because it looks like the assertion returns false but doesn't fail the test. Which is what the issue describes as well. Oh well :)

Reply:
I just tried it by changing the counter to not count up. Then the test fails.